### ISSUE TITLE
New trainer doesn't train the model frame by frame

### ISSUE THREAD

**Issue by quetalasj** [quetalasj/imap-pytorch#1]
- [x] #2

**Comment by quetalasj** [comment]
The trainer can't train if the frames are not shuffled

**Comment by quetalasj** [comment]
- [x] Check if pixels randomization improves the situation

**Comment by quetalasj** [comment]
Random shuffling the pixels in each frame improves the situation

**Comment by quetalasj** [comment]
- [x] Check if more frequently switching images improves the quality in training step
It improves the quality (3072 points per image)   

Outcomes:
1. Model can't be trained if batches have the sequential pixels of one frame (`shuffle=False`)
2. BUT Model can be trained (with `shuffle=False`) if batches have the random pixels of one frame
3. AND Model can be improved if it is trained on a small number of random pixels for each image
4. Parallel data loading (`num_workers>1`) has undefined behavior in the current implementation

**Comment by quetalasj** [comment]
### **Lighting trainer with all shuffled images result:**
![image](https://user-images.githubusercontent.com/34664976/148936373-3363cbeb-8140-43b5-a2fc-9b1aa6a761bb.png)
### **New Trainer with random pixels and sequential images result (2000 epochs, 3072 pixels per image, 20 minutes of training)**
![image](https://user-images.githubusercontent.com/34664976/148936722-f2340921-1104-48ba-9f38-17d50b3dbc11.png)
